packages/material-ui/src/Fade/Fade.js

```diff
@@ -60,16 +62,19 @@ function Fade(props) {
             ...style,
             ...children.props.style,
           },
+          ref: handleRef,
           ...childProps,
         });
       }}
     </Transition>
   );
-}
+});
 
 Fade.propTypes = {
   /**
    * A single child content element.
+   *
+   * ⚠️The component used as a child [must be able to hold a ref](/guides/composition/#children).
```

Comment:
Do we need this warning? We don't use the ref internally. It's for the users.

Reply:
I don't mind if we don't have this here. With TrapFocus requiring a ref, everything inside a Modal will need to be able to hold a ref. Maybe it's better to put a warning there instead?

Reply:
No, what I mean is that the component does need the ref. It's all user base.

Reply:
Maybe all components that have children should forward refs to them by default?

Reply:
I think that we should forward the ref where we forward the props. This strategy should be future proof as raised by Sebastian.

Reply:
I only mentioned children as an example but just meant forwarding by default :)

Reply:
Not sure what you mean. We should have ref forwarding by default. At least that's the goal. Only the transition components don't have that yet because it wasn't needed previously and without useForkRef it's not safe to claim other refs.

The only exception is when we actually need an imperative handle. TouchRipple is an example. We call e.g. `start` imperatively in the ButtonBase component.